lib/quda_arpack_interface.cpp

```diff
@@ -88,18 +96,17 @@ namespace quda
     case QUDA_SPECTRUM_LI_EIG: strcpy(spectrum, "LI"); break;
     default: errorQuda("Unexpected spectrum type %d", eig_param->spectrum);
     }
-
-    bool reverse = true;
-    if (strncmp("L", spectrum, 1) == 0 && !eig_param->use_poly_acc) {
-      reverse = false;
-    } else if (strncmp("S", spectrum, 1) == 0 && eig_param->use_poly_acc) {
-      reverse = false;
+    
+    bool reverse = false;
+    if (strncmp("S", spectrum, 1) == 0 && eig_param->use_poly_acc) {
+      // Smallest eig requested by use, largest will requested from ARPACK
+      // due to poly acc
+      reverse = true;
       spectrum[0] = 'L';
-    } else if (strncmp("L", spectrum, 1) == 0 && eig_param->use_poly_acc) {
-      reverse = false;
-      spectrum[0] = 'S';
     }
-
+    
```

Comment:
is this a left over debug statement @cpviolator ?

Reply:
I realised that this is the only time one needs to intervene to get the eigenvalues in ascending order with respect to the sorted Ritz values. For LR and SR with no poly-acc, the Ritz and eigenvalues will be in the same order. With poly acc and requesting SR, the Ritz are descending and the eigenvalues are ascending. Just a statement to let the user know what's going on.

Reply:
Ok, well I would suggest that this be guarded with `>= QUDA_VERBOSE`.

Reply:
I removed it, no need for it really.